tests/integration/test_api/test_config/test_logs/test_api_logs_formats.py

```diff
@@ -0,0 +1,192 @@
+                          wait_for_start):
+    '''
+    description: Check if the logs of the API are stored in the specified formats and the content of the log
+                 files are the expected.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_api_environment:
+            type: fixture
+            brief: Configure a custom environment for API testing.
+        - clean_log_files:
+            type: fixture
+            brief: Reset the log files of the API and delete the rotated log files.
+        - daemons_handler:
+            type: fixture
+            brief: Handle the Wazuh daemons.
+        - wait_for_start:
+            type: fixture
+            brief: Wait until the API starts.
+    assertions:
+        - Verify that the expected log exists in the log file.
+        - Verify that the values of the log are the same in both log formats.
+    input_description: The test gets the configuration from the YAML file, which contains the API configuration.
+    expected_output:
+        - The log was not expected.
+        - The length of the subgroups of the match is not equal.
+        - The values of the logs don't match.
+    tags:
+        - api
+        - logs
+        - logging
+    '''
+    wazuh_log_monitor = FileMonitor(API_LOG_FILE_PATH)
+    json_log_monitor = FileMonitor(API_JSON_LOG_FILE_PATH)
+
+    current_formats = get_configuration['configuration']['logs']['format'].split(',')
+    current_level = get_configuration['configuration']['logs']['level']
+    send_request()
+
+    expected_error =  'expected_error' in get_configuration
+    if 'json' in current_formats:
+        callback = callback_json_log_error if expected_error else callback_json_log_login_info
+        json_result = json_log_monitor.start(timeout=API_GLOBAL_TIMEOUT, callback=callback,
+                                             error_message=f'JSON API {current_level} log was not been generated.'
+                                             ).result()
+    if 'plain' in current_formats:
+        callback = callback_plain_error if expected_error else callback_plain_log_login_info
+        plain_result = wazuh_log_monitor.start(timeout=API_GLOBAL_TIMEOUT, callback=callback,
+                                               error_message=f'Plain API {current_level} log was not the expected.'
+                                               ).result()
+    if len(current_formats) == 2:
+        assert len(json_result.groups()) == len(plain_result.groups()), 'The length of the subgroups of the match is ' \
+                                                                        'not equal.' \
+                                                                        'Subgroups of the JSON match:' \
+                                                                        f' {len(json_result.groups())}\n' \
```

Comment:
```suggestion
                                                                        f" {len(json_result.groups())}\n" \
```

Reply:
ok. fixed.

Reply:
Done in ae04711e46bfb2b5dedbf08c5fafc046a9b4a61b